### PR TITLE
Pack10

### PR DESCRIPTION
**Pull request by funman**

### CODE REVIEW

**Comment by funman** [comment]
This should go after ubuf_append branch, and I also need to disable asm build for non x86

File: lib/upipe-hbrmt/upipe_unpack10bit.c [hunk 1]
```diff
@@ -111,6 +113,7 @@ static bool upipe_unpack10bit_handle(struct upipe *upipe, struct uref *uref,
     if (unlikely(ubase_check(uref_flow_get_def(uref, &def)))) {
         upipe_unpack10bit_store_flow_def(upipe, NULL);
         upipe_unpack10bit_require_ubuf_mgr(upipe, uref);
+        uref_block_flow_set_align(uref, UBUF_ALIGN);
```
Comment (cmassiot) [inline lib/upipe-hbrmt/upipe_unpack10bit.c]: I think you want to move it up one line.

File: lib/upipe-hbrmt/upipe_pack10bit.c [hunk 2]
```diff
@@ -0,0 +1,389 @@
...
+ * @param command type of command to process
+ * @param args arguments of the command
+ * @return an error code
+ */
+static int upipe_pack10bit_control(struct upipe *upipe, int command, va_list args)
+{
+    switch (command) {
+        case UPIPE_REGISTER_REQUEST: {
+            struct urequest *request = va_arg(args, struct urequest *);
+            return upipe_pack10bit_alloc_output_proxy(upipe, request);
```
Comment (cmassiot) [inline lib/upipe-hbrmt/upipe_pack10bit.c]: You shouldn't forward the ubuf_mgr and flow_format requests, as you are reallocating the ubufs.

File: lib/upipe-hbrmt/upipe_unpack10bit.c [hunk 3]
```diff
@@ -0,0 +1,368 @@
...
+ * @param upipe description structure of the pipe
+ * @param command type of command to process
+ * @param args arguments of the command
+ * @return an error code
+ */
+static int upipe_unpack10bit_control(struct upipe *upipe, int command, va_list args)
+{
+    switch (command) {
+        case UPIPE_REGISTER_REQUEST: {
+            struct urequest *request = va_arg(args, struct urequest *);
```
Comment (cmassiot) [inline lib/upipe-hbrmt/upipe_unpack10bit.c]: Same here.

**Pull request merged**